wagtailsharing/tests/test_models.py

```diff
@@ -84,3 +86,35 @@ def test_root_url_port_443_https(self):
     def test_root_url_other_port_http(self):
         site = SharingSite(hostname="test.hostname", port=1234)
         self.assertEqual(site.root_url, "http://test.hostname:1234")
+
+
+class TestSharableRoutablePage(TestCase):
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.default_site = Site.objects.get(is_default_site=True)
+        self.routable_page = self.default_site.root_page.add_child(
+            instance=ShareableRoutablePage(title="routable")
+        )
+        self.routable_page.save_revision().publish()
+
+    def test_route_no_draft(self):
+        request = self.factory.get("/")
+
+        with mock.patch.object(
+            ShareableRoutablePage, "get_latest_revision_as_page"
+        ) as mock_get_latest_revision_as_page:
+            self.routable_page.route(request, [])
+
+        mock_get_latest_revision_as_page.assert_not_called()
+
+    def test_route_with_draft(self):
+        request = self.factory.get("/")
+        setattr(request, "routed_by_wagtail_sharing", True)
+
+        with mock.patch.object(
+            ShareableRoutablePage, "get_latest_revision_as_page"
+        ) as mock_get_latest_revision_as_page:
+            mock_get_latest_revision_as_page.return_value = self.routable_page
+            self.routable_page.route(request, [])
+
+        mock_get_latest_revision_as_page.assert_called()
```

Comment:
I find it pretty difficult to understand what these are testing. Would it be possible to create a test routable page model that has some different routes and verify via test that the routing works?

Reply:
I'm trying to test that the `get_latest_revision` is called on the page before routing. 

I could create a test routable page, but that would add significant overhead to the tests (it'd require creating a test app that could house the model, etc). I wasn't sure that was worth it here. It's certainly doable.

Reply:
32494ac adds a test app and removes the mocking here. The full request cycle is now tested.